js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,347 @@
+import Alert from '../../common/components/Alert';
+import Button from '../../common/components/Button';
+import LinkButton from "../../common/components/LinkButton";
+import Page from '../../common/components/Page';
+import Select from '../../common/components/Select';
+import Switch from '../../common/components/Switch';
+import icon from '../../common/helpers/icon';
+import listItems from '../../common/helpers/listItems';
+import ItemList from '../../common/utils/ItemList';
+import Stream from '../../common/utils/Stream';
+import LoadingModal from './LoadingModal';
+import ExtensionPermissionGrid from './ExtensionPermissionGrid';
+import saveSettings from '../utils/saveSettings';
+
+export default class ExtensionPage extends Page {
+  oninit(vnode) {
+    super.oninit(vnode);
+
+    this.loading = false;
+    this.extension = app.data.extensions[this.attrs.id];
+    this.changingState = false;
+    this.settings = {};
+
+    // Backwards compatibility layer will be removed in
+    // Beta 16
+    if (app.extensionSettings[this.extension.id]) {
+      app.extensionData[this.extension.id] = app.extensionSettings[this.extension.id];
+    }
+  }
+
+  className() {
+    return this.extension.id + '-Page';
+  }
+
+  view() {
+    return (
+      <div className={'ExtensionPage ' + this.className()}>
+        {this.header()}
+        {!this.isEnabled() ? (
+          <div className="container">
+            <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.enable_to_see')}</h2>
+          </div>
+        ) : (
+          <div className="ExtensionPage-body">{this.sections().toArray()}</div>
+        )}
+      </div>
+    );
+  }
+
+  content() {
+    const settings = this.getContent('settings');
+
+    if (settings) {
+      const items = new ItemList();
+
+      Object.keys(settings).map((key) => {
```

Comment:
I think this would be great as a util, so it's easier to use standard controls in custome extension pages

Reply:
This problem with that is we need this to be in the context of the extensions page. The can easily use that still by just extending the base extension page.

Reply:
Maybe as a method then?

Reply:
I think we need to pick whether to combine everything into content or to split the settings section and settings content. We are getting to extendable levels that aren't necessary, adding many lines of code to save developers literally 1 line.

Reply:
I think the content method should return jsx. That means that if someone is replacing content entirely but still wants to use this útil for converting settings descriptors to actual components they still could.

Reply:
So maybe move the util to a different function in the settings page?